cmd/kube-proxy/app/server_others.go

```diff
@@ -329,6 +336,49 @@ func (s *ProxyServer) setupConntrack() error {
 		}
 	}
 
+	if s.Config.Conntrack.UDPTimeout.Duration > 0 {
+		timeout := int(s.Config.Conntrack.UDPTimeout.Duration / time.Second)
+		if err := ct.SetUDPTimeout(timeout); err != nil {
+			return err
+		}
+	}
+
+	if s.Config.Conntrack.UDPStreamTimeout.Duration > 0 {
+		timeout := int(s.Config.Conntrack.UDPStreamTimeout.Duration / time.Second)
+		if err := ct.SetUDPStreamTimeout(timeout); err != nil {
+			return err
+		}
+	}
+
+	if s.Config.Conntrack.TCPBeLiberal != nil {
+		if *s.Config.Conntrack.TCPBeLiberal {
+			if err := ct.SetTCPBeLiberal(1); err != nil {
+				return err
+			}
+		}
+	} else {
+		// try setting nf_conntrack_tcp_be_liberal=1 if kernelVersion < CTStateInvalidFixKernelVersion
+		// don't return any error, just log them.
+		kernelVersionStr, err := ipvs.NewLinuxKernelHandler().GetKernelVersion()
+		if err != nil {
+			klog.ErrorS(err, "failed to get kernel version")
+			return nil
+		}
+
+		kernelVersion, err := version.ParseGeneric(kernelVersionStr)
+		if err != nil {
+			klog.ErrorS(err, "error parsing kernel version", "kernelVersionStr", kernelVersionStr)
+			return nil
+		}
+
+		if kernelVersion.LessThan(version.MustParseGeneric(CTStateInvalidFixKernelVersion)) {
+			klog.InfoS("SetupConntrack: kernel is not patched with netfilter conntrack INVALID state bug, setting nf_conntrack_tcp_be_liberal=1", "kernelVersion", kernelVersionStr, "fixedKernelVersion", CTStateInvalidFixKernelVersion)
```

Comment:
There will be kernels with version numbers less than 6.1 that have the fix. (It's already been backported to RHEL 8 and 9, and maybe other LTS kernels too.) Maybe don't assert that the current kernel definitely has the bug, just say something like "Setting nf_conntrack_tcp_be_liberal=1 to work around conntrack bugs in older kernels"?